File: euler/problem7.js
/*
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that 
the 6th prime is 13.

What is the 10 001st prime number?
*/


console.log(nthPrime(parseInt(process.argv[2])));


function nthPrime(nth){
  var primes = [2], next = 3, div, i, limit, isPrime;
  while(primes.length<nth){
    div = 3; 
    i = 1;
    limit= Math.sqrt(next)+1;
    isPrime= true;
    while(div<limit){
      if(next%div === 0){
          isPrime = false;
          div = limit;
      }
      else {
        div = primes[i++] || div+ 2;
      }
    }
    if(isPrime){
      primes.push(next);
    }
    next += 2;
  }
  return primes[primes.length-1];
}

File: euler/problem7.py
'''
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

What is the 10 001st prime number?
'''

# def nthPrime(n):
#   count = 1
#   primes = [2, 3]
#   while(count < n):
#     current = primes[len(primes)-1] + 2
#     index = 0
#     divisor = primes[index]
#     print current
#     count += 1
#   print primes

# nthPrime(2) nthPrime(3)

li = [1,2,3]

#TODO

try: 
  print li[3]
except IndexError:
  print 'Error!'
